bootstrap_cfn/iam.py

```diff
@@ -18,18 +20,299 @@ def __init__(self, aws_profile_name, aws_region_name='eu-west-1'):
 
     def upload_ssl_certificate(self, ssl_config, stack_name):
         for cert_name, ssl_data in ssl_config.items():
-            cert_body = ssl_data['cert']
-            private_key = ssl_data['key']
-            try:
-                cert_chain = ssl_data['chain']
-            except:
-                cert_chain = None
-            cert_id = "{0}-{1}".format(cert_name, stack_name)
-            self.conn_iam.upload_server_cert(cert_id, cert_body, private_key, cert_chain)
+            self.upload_certificate(cert_name,
+                                    stack_name,
+                                    ssl_data,
+                                    force=True)
         return True
 
     def delete_ssl_certificate(self, ssl_config, stack_name):
-        for cert_name in ssl_config.keys():
-            cert_id = "{0}-{1}".format(cert_name, stack_name)
-            self.conn_iam.delete_server_cert(cert_id)
+        for cert_name, ssl_data in ssl_config.items():
+            self.delete_certificate(cert_name,
+                                    stack_name,
+                                    ssl_data,
+                                    force=True)
         return True
+
+    def update_ssl_certificates(self, ssl_config, stack_name, force=False):
+        """
+        Update all the ssl certificates in the identified stack
+
+        Args:
+            ssl_config(dictionary): A dictionary of ssl configuration data organised by
+                cert_name to a dictionary with the config data in it
+            stack_name(string): The name of the stack
+            force(bool): Update no whether required or not if True, only update
+                if certificates differ if False
+        Returns:
+            succeeded_count(int): Number of certificates that were successfully updated
+        """
+        succeeded_count = 0
+        for cert_name, ssl_data in ssl_config.items():
+            try:
+                cert_exists = self.check_remote_certificate_exists(cert_name,
+                                                                   stack_name,
+                                                                   ssl_data)
+                if cert_exists:
+                    if force:
+                        self.delete_certificate(cert_name,
+                                                stack_name,
+                                                ssl_data,
+                                                force)
+
+                    upload_success = self.upload_certificate(cert_name,
+                                                             stack_name,
+                                                             ssl_data,
+                                                             force)
```

Comment:
Oh I see, it's ok because the cert your deleting remains attached to the listener on the ELB?